README.md

```diff
@@ -31,86 +31,104 @@ Umbra is a protocol for enabling stealth payments on the Ethereum blockchain. It
+2. The corresponding public keys are both published on-chain as records associated with your ENS or CNS name.
+3. A payer uses your ENS or CNS name to look up your two public keys. Separately, the payer generates a random number.
+4. The random number is used with the spending public key to generate a "stealth address" to send funds to. The same random number is used with the viewing public key to encrypt the random number.
+5. Using the Umbra contract, the payer sends funds to the stealth address and the encrypted data is emitted as an Announcement event.
+6. The receiver scans all Announcement events from the Umbra contract. For each, they use their viewing private key to decrypt the random number, then multiply that number by their spending private key to generate the stealth private key. If the stealth private key controls the address funds were sent to, this payment was for the receiver
+7. The receiver can now use the private key to either directly send the transaction required to withdraw funds to another address, or sign a meta-transaction to have the withdrawal request processed by a relayer.
 
-Umbra is meant for payments between two entities, and comes with a different set of privacy tradeoffs. Rather than *breaking* the link between sending and receiving address, Umbra makes that link *meaningless*. Everyone knows the address funds were sent to, but they don't know who controls that address.
+See the [Technical Details: How does it work?](https://app.umbra.cash/faq#how-does-it-work-technical) for more details.
 
-**What advantages does Umbra have?**
+#### How private is Umbra?
 
-* Umbra allows arbitrary amounts to be sent, since there is no need for inputs and outputs to be uniform.
-* Umbra does not require the receiver to wait to withdraw funds— as soon as they're sent, they can be withdrawn to any address the receiver chooses.
-* Umbra ensures only the receiver can withdraw the funds once they're sent. The sender does not hold the private key of the receiving address.
-* Umbra uses significantly less gas, as it does not require the verification of any advanced cryptography on chains. All transactions are simple transfers.
-* Umbra enables ETH and arbitrary ERC20 tokens to be transferred privately. You're not dependent on a large anonymity set developing for each token.
+Umbra offers a limited set of privacy guarantees and it’s important to understand them before using the protocol. Umbra does not offer "full" privacy like Aztec or Zcash. It simply makes it impossible for any outside observers (i.e. anyone who is not the sender or the receiver) to know who the sender paid by looking at the receiving address.
 
-**Can you give an example of a practical use of Umbra?**
+It’s important to understand that poor hygiene by the receiver— for example, sending the funds directly to a publicly known address— eliminates the privacy benefits for both the sender and receiver.
```

Comment:
Maybe "reduces" instead of "eliminates", because you can claim you withdrew funds to someone else's address, i.e. poor hygiene means it's likely, but not guaranteed, the stealth address owner is also the withdrawal address owner
```suggestion
It’s important to understand that poor hygiene by the receiver— for example, sending the funds directly to a publicly known address— reduces the privacy benefits for both the sender and receiver.
```